### PR TITLE
Add spire-lib chart

### PR DESCRIPTION
**Pull request by kfox1111**
Make spire-lib bits into its own library chart.

### CODE REVIEW

**Comment by kfox1111** [comment]
Squashed.

File: charts/spire/templates/NOTES.txt [hunk 1]
```diff
@@ -1,35 +1 @@
...
-{{-     if (index .Values "spire-server").controllerManager.watchClassless }}
-
-Spire CR's will be handled if no className is specified or if className is set to "{{ $className }}"
-{{-     else }}
-
-Spire CR's will be handled only if className is set to "{{ $className }}"
-{{-     end }}
-{{-   end }}
-{{- end }}
+{{ include "spire-lib.notes" . }}
```
Comment (marcofranssen) [inline charts/spire/templates/NOTES.txt:1]: Wouldn't each chart define its own Notes specific to that chart?

E.g. once we break out the spire-agent, and other charts, each of those charts would have their own specific notes, then this reusable Notes thing seems a bit overkill as you would have to do all kind of if statements.

File: charts/spire/charts/spire-lib/values.yaml
Comment (marcofranssen) [inline charts/spire/charts/spire-lib/values.yaml:1]: Can't we remove this dummy file?

File: charts/spire/charts/spire-lib/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.1.0
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: "0.1.0"
```
Comment (marcofranssen) [inline charts/spire/charts/spire-lib/Chart.yaml:24]:
```suggestion
appVersion: ""
```

the appVersion can be left empty. As the lib chart probably will not be dependant on a specific spire version and isn't installing an app I suggest to leave it empty.

**Review by marcofranssen — COMMENTED** [review]
Some inline feedback.

File: charts/spire/templates/NOTES.txt (hunk 1 above)
Comment (kfox1111) [reply charts/spire/templates/NOTES.txt:1]: Yeah. I thought about that. Charts that look very similar though could use the same notes most likely. Those that deviate too much from the base chart could do their own notes and ignore the macro entirely though?

File: charts/spire/charts/spire-lib/values.yaml
Comment (kfox1111) [reply charts/spire/charts/spire-lib/values.yaml:1]: No. The doc tool flips out. :/

File: charts/spire/templates/NOTES.txt (hunk 1 above)
Comment (kfox1111) [reply charts/spire/templates/NOTES.txt]: I reverted the notes bit.

Comment (faisal-memon) [reply charts/spire/templates/NOTES.txt]: @marcofranssen Any further comments on this pr?

**Pull request merged**